aptos-move/framework/aptos-framework/sources/transaction_fee.move

```diff
@@ -19,9 +108,193 @@ module aptos_framework::transaction_fee {
         );
     }
 
+    /// Burn a collected transaction fee.
+    public(friend) fun burn_collected_fee(coin: Coin<AptosCoin>) acquires AptosCoinCapabilities {
```

Comment:
This function is redundant with burn_coin_fraction. It's basically the same as burn_coin_fraction with 100%. IMO, it's cleaner if we:

1. Have a single burn_coin function that takes a percentage.
2. Delete burn_collected_fee.
3. Where burn_collected_fee is called, call burn_coin(100%). Replace burn_coin_fraction with burn_coin